src/InclusiveScans.jl

```diff
@@ -1,12 +1,188 @@
 module InclusiveScans
 
+using CUDA
 
-"""
-    hi = hello_world()
-A simple function to return "Hello, World!"
-"""
-function hello_world()
-    return "Hello, World!"
+# CUDA scan block size
+const BLOCK_SIZE = 1024
+
+@inline function _ceil_div(a, b)
+    return cld(a, b)
+end
+
+# Execute BLELLOCH EXCLUSIVE‐prefix-sum for current block.
+# - Each block processes 2 * BLOCK_SIZE elements (2 per thread).
+# - If `blockSums !== nothing`, writes the "block sum" to blockSums[blockIdx.x].
+# - Otherwise skips this step (when we scan blockSums itself).
+function _scanBlockKernel!(
+    g_odata::CuDeviceVector{Float32,1},
+    g_idata::CuDeviceVector{Float32,1},
+    blockSums::Union{CuDeviceVector{Float32,1},Nothing},
+    n::Int,
+)
+    # Dynamic shared memory on 2*BLOCK_SIZE float
+    temp = @cuDynamicSharedMem(Float32, blockDim().x * 2)
+
+    tx = threadIdx().x - 1
+    bx = blockIdx().x - 1
```

Comment:
Consider using `Int32` here. In Julia `1` is an `Int64` per default.

Reply:
https://cuda.juliagpu.org/stable/tutorials/performance/#32-bit-Integers